smoker/server/plugins/__init__.py

```diff
@@ -461,6 +463,14 @@ def __init__(self, name, queue, params, forced=False):
 
     def run(self):
         setproctitle.setproctitle('smokerd plugin %s' % self.plugin_name)
+
+        # close unnecessary open sockets created while cloning the subprocess
+        open_sockets = filter(lambda x: type(x) == socket._socketobject, gc.get_objects())
+        for s in open_sockets:
+            if s.type != 2:
+                #lg.info("Closing socket %s in plugin %s" % (s.getsockname(), self.plugin_name))
+                s.close()
+
```

Comment:
maybe this code would be worth a nice separate function 'close_unecessary_sockets', similar to the 'drop_privileges'